corehq/util/compression.py

```diff
@@ -9,10 +10,9 @@ def decompress(compressed):
 
     # Build the dictionary.
     dict_size = 0x10000
-    dictionary = dict((unichr(i), unichr(i)) for i in range(dict_size))
-    # in Python 3: dictionary = {chr(i): chr(i) for i in range(dict_size)}
+    dictionary = dict((six.unichr(i), six.unichr(i)) for i in range(dict_size))
 
-    result = StringIO()
+    result = io.StringIO()
```

Comment:
Looks like this is going from cStringIO to StringIO instead of BytesIO. What's the methodology that you use to pick out which to use?

Reply:
I don't have a good methodology - I inspect each instance, decide whether it deals with unicode or bytes, and test if I'm unsure.  It's a slow going process 🐢